### DISCUSSION TITLE
Profiling llama on AMD backend, found  long times CopyHostToDevice

### DISCUSSION THREAD

**Discussion by CrimsonDump** [vllm-project/vllm#4855] [Q&A]
I've been running llama 2 with vllm, and trace with rocprof. This is what I got,  ~8 ms CopyHostToDevice CUDA time in a ~ 12 ms deocode phase, though these copies seem to be harmless to the core kernels (paged_attention, gemm, etc.. ). Then I found some clues from tracing which suggest that these CopyHostToDevice is for computing logits in Sampler, which confuses me because I thought all Weight were loaded at very beginning (nv did), instead of each decode phase.

So I'm wondering what are these CopyHostToDevice doing? Is it true sampler need to load weight every decode phase? 

<img width="762" alt="detail" src="https://github.com/ROCm/ROCm/assets/56749892/d14d7cbe-64f3-4a7e-bb3b-a51b1c9f68ad">
<img width="580" alt="one_decode" src="https://github.com/ROCm/ROCm/assets/56749892/4a738e84-2586-47eb-bea3-dead97f95070">
<img width="950" alt="detail_args" src="https://github.com/ROCm/ROCm/assets/56749892/3e9f77e3-0ae5-4aa2-a8a6-3a1a8da2fd79">
<img width="580" alt="code" src="https://github.com/ROCm/ROCm/assets/56749892/b9d6eb29-cacb-496d-9991-e27855d3eab6">
